pkg/proc/gdbserial/gdbserver_conn.go

```diff
@@ -796,8 +798,12 @@ func (conn *gdbConn) parseStopPacket(resp []byte, threadID string, tu *threadUpd
 				sp.watchAddr = medata[1] // this should be zero if this is really a single step stop and non-zero for watchpoints
 			}
 		case "arm64":
-			if metype == _EXC_BREAKPOINT && len(medata) >= 2 && medata[0] == _EXC_ARM_DA_DEBUG {
-				sp.watchAddr = medata[1]
+			if metype == _EXC_BREAKPOINT && len(medata) >= 2 && (medata[0] == _EXC_ARM_DA_DEBUG || medata[0] == _EXC_I386_SGL) {
+				if medata[1] <= 6 {
```

Comment:
I need to confirm that the HWbreakpoint index is 6 max, or if there's a better way to make this determination.